File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    let top = 0;
    let time = 0;
    
    while(top < progresses.length) {
        time++;
        let cur = 0;
        
        for (let i = top; i < progresses.length; i++) {
            if (progresses[i] + speeds[i] * time < 100) {
                break;
            }
            
            cur++;
        }
        
        if (cur) {
            top += cur;
            answer.push(cur);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    left_days = [math.ceil((100 - progresses[i]) / speeds[i]) for i in range(len(progresses))]
    cur_day = left_days[0]
    head = 0
    answer = []

    while head < len(left_days):
        temp = 0
        for i in range(head, len(left_days)):
            if cur_day < left_days[i]:
                cur_day = left_days[i]
                break
            
            temp += 1
            head += 1
        
        if temp > 0:
            answer.append(temp)
    
    return answer
